### PR TITLE
Added 8 new  and updated 2 subscription controls

### PR DESCRIPTION
**Pull request by pranchalsomani**

### CODE REVIEW

File: Control coverage/Feature/SubscriptionCore.md [hunk 1]
```diff
@@ -1201,4 +1246,479 @@ SPNs have a single credential and most scenarios that use them cannot support mu
...
+
+### Rationale 
+SPNs having access to subscription must have secrets within maximum approved expiry time.
+
+### Control Settings 
+```json 
+{
+    "ExpirationPeriodInDays": 380,
+    "ServicePrincipalTypeFilter": [ "Application", "Legacy" ],
+    "AllowedObjectIds": []
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:1564]: remove AllowedObjectIds as it has empty value

File: Control coverage/Feature/SubscriptionCore.md [hunk 2]
```diff
@@ -1201,4 +1246,479 @@ SPNs have a single credential and most scenarios that use them cannot support mu
...
+### Recommendation 
+
+- **Azure Portal** 
+    
+    To enable this plan on all servers in your subscription: From Defender for Cloud's 'Environment settings' page, select the relevant subscription. In the 'Defender plans' page, set 'Servers' to 'On'."
+
+
+### Azure Policies or REST APIs used for evaluation 
+
+- API to get providers list with registration status for subscription level : - /subscriptions/{subscriptionId}/providers?api-version=2020-06-01&$select=namespace,registrationstate <br />
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:1294]: API to get providers -> Rest API to get providers

File: Control coverage/Feature/SubscriptionCore.md [hunk 3]
```diff
@@ -1201,4 +1246,479 @@ SPNs have a single credential and most scenarios that use them cannot support mu
...
+### Recommendation 
+
+- **Azure Portal** 
+
+    To enable this plan on all containers in your subscription: From Defender for Cloud's 'Environment settings' page, select the relevant subscription --> In the 'Defender plans' page, set 'Containers' to 'On'
+
+
+### Azure Policies or REST APIs used for evaluation 
+
+- API to get providers list with registration status for subscription level : - /subscriptions/{subscriptionId}/providers?api-version=2020-06-01&$select=namespace,registrationstate <br />
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:1351]: API to get providers -> Rest API to get providers

File: Control coverage/Feature/SubscriptionCore.md [hunk 4]
```diff
@@ -1201,4 +1246,479 @@ SPNs have a single credential and most scenarios that use them cannot support mu
...
+### Recommendation 
+
+- **Azure Portal** 
+    
+    To enable this plan on all key vaults in your subscription: From Defender for Cloud's 'Environment settings' page, select the relevant subscription --> In the 'Defender plans' page, set 'Key Vault' to 'On'
+
+
+### Azure Policies or REST APIs used for evaluation 
+
+- API to get providers list with registration status for subscription level : - /subscriptions/{subscriptionId}/providers?api-version=2020-06-01&$select=namespace,registrationstate <br />
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:1408]: Please correct at all applicable places, API to get providers -> Rest API to get providers

File: Control coverage/Feature/SubscriptionCore.md [hunk 5]
```diff
@@ -438,18 +446,39 @@ Permanent access increase the risk of a malicious user getting that access and i
 ### Control Settings
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:446]: Change the display name for this control to  **Do not grant permanent access for Subscription level roles**

File: Control coverage/Feature/SubscriptionCore.md [hunk 6]
```diff
@@ -458,10 +487,10 @@ Permanent access increase the risk of a malicious user getting that access and i
 ### Control Spec 
 
 > **Passed:** 
-> No critical permanent role (Owner, Contributor, UAA) assignments present at subscription level OR if no role assignment present in subscription
+> No permanent role assignments present at subscription level apart from explicitly allowed roles OR if no role assignment present in subscription.
 > 
 > **Failed:** 
-> Any critical permanent role (Owner, Contributor, UAA) assignments present at subscription level
+> Any critical permanent role assignments present at subscription level apart from explicitly allowed roles.
```
Comment (v-rahkuma) [inline Control coverage/Feature/SubscriptionCore.md:493]: Any **critical** permanent role assignments -> Any permanent role assignments

File: Control coverage/Feature/SubscriptionCore.md (hunk 5 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:446]: Fixed

File: Control coverage/Feature/SubscriptionCore.md (hunk 6 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:493]: Fixed.

File: Control coverage/Feature/SubscriptionCore.md (hunk 2 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:1294]: Fixed.

File: Control coverage/Feature/SubscriptionCore.md (hunk 3 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:1351]: Fixed.

File: Control coverage/Feature/SubscriptionCore.md (hunk 4 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:1408]: Done.

File: Control coverage/Feature/SubscriptionCore.md (hunk 1 above)
Comment (pranchalsomani) [reply Control coverage/Feature/SubscriptionCore.md:1564]: Fixed

**Review by Aboli-msft — APPROVED** [review]

**Review by v-rahkuma — APPROVED** [review]

**Pull request merged**